File: RCScommon/ToDb.m
function [dB_VV, dB_HH, s1, s2] = ToDb(EEVV, EEHH,RangeFreqThetaPhi,newteta,newphi,newfrequency,a,Nphi,Nteta,Nfreq) %#codegen
%ToDb RangeProfile, RCS_vs_Freq, RCS_vs_Theta and RCS_vs_Phi
% E_VV, E_HH, E_VVt, E_HHt converts to db
% EEVV, EEHH comes from "[a, freq, EEVV, EEHH] = RCSRangeCommon(dosyaAdi)" function
% RangeFreqThetaPhi is switch between mods.
if nargin<4
    newteta = 1;
elseif nargin<5
    newphi = 1;
elseif nargin<6
    newfrequency = 1;
end
if coder.target('Matlab')
    if RangeFreqThetaPhi>5 && RangeFreqThetaPhi<1
        error('RangeFreqThetaPhi: 1->RangeProfile\n2->RCS_vs_Freq\n3->RCS_vs_Theta\n4->RCS_vs_Phi')
    end
end

if RangeFreqThetaPhi == 1
    % RangeProfile
    E_VV=EEVV(:,newteta,newphi);
    E_VVt=fft(E_VV);
    dB_VVtt = todbsmall(E_VVt); % dB_VVt
    dB_VV = fftshift(dB_VVtt);
    
    E_HH=EEHH(:,newteta,newphi);
    E_HHt=fft(E_HH);
    dB_HHtt = todbsmall(E_HHt); % dB_HHt
    dB_HH = fftshift(dB_HHtt);
    
    s1=a(1:Nphi,3);
    s2=a(1:Nphi:Nteta*Nphi,2);
elseif RangeFreqThetaPhi == 2
    % RCS_vs_Freq
    E_VV=EEVV(:,newteta,newphi);
    dB_VV = todbsmall(E_VV);
    
    E_HH=EEHH(:,newteta,newphi);
    dB_HH = todbsmall(E_HH);
    
    s1=a(1:Nphi,3);
    s2=a(1:Nphi:Nteta*Nphi,2);
elseif RangeFreqThetaPhi == 3
    %RCS_vs_Theta
    E_VV=EEVV(newfrequency,:,newphi);
    dB_VV = todbsmall(E_VV);
    
    E_HH=EEHH(newfrequency,:,newphi);
    dB_HH = todbsmall(E_HH);
    
    s1=a(1:Nphi*Nteta:Nfreq*Nteta*Nphi,1);
    s2=a(1:Nphi,3);
elseif RangeFreqThetaPhi == 4
    % RCS_vs_Phi
    E_VV=EEVV(newfrequency,newteta,:);
    dB_VV = todbsmall(E_VV);
    dB_VV = dB_VV(1,:);
    
    E_HH=EEHH(newfrequency,newteta,:);
    dB_HH = todbsmall(E_HH);
    dB_HH = dB_HH(1,:);
    
    s1=a(1:Nphi*Nteta:Nfreq*Nteta*Nphi,1);
    s2=a(1:Nphi:Nteta*Nphi,2);
else % error contidition
    dB_VV = 0; dB_HH = 0; s1= 0; s2 = 0;
end



end

File: RCScommon/versiyonpython/ToDb.py
# from libsmop import * # bu modülü kaldırmak için düzenlemeyle numpy fonksiyonlarıyla değiştir.
from todbsmall import todbsmall

import numpy as np
from numpy.core._multiarray_umath import ndarray


def ToDb(eevv: ndarray = None, eehh: ndarray = None, range_freq_theta_phi: int = None, newteta: int = 1, newphi: int = 1,
         newfrequency: int = 1, a: ndarray = None, Nphi: int = None, Nteta: int = None, Nfreq: int = None) -> tuple:
    # ToDb RangeProfile, RCS_vs_Freq, RCS_vs_Theta and RCS_vs_Phi
    # e_vv, e_hh, E_VVt, E_HHt converts to db
    # eevv, eehh comes from "[a, freq, eevv, eehh] = RCSRangeCommon(dosyaAdi)" function
    # range_freq_theta_phi is switch between mods.
    if 5 < range_freq_theta_phi < 1:
        import sys
        print('range_freq_theta_phi: 1->RangeProfile\n2->RCS_vs_Freq\n3->RCS_vs_Theta\n4->RCS_vs_Phi', file=sys.stderr)

    if range_freq_theta_phi == 1:
        # RangeProfile
        e_vv = eevv[:, newteta-1, newphi-1]  # ToDb.m:21
        E_VVt = np.fft.fft(e_vv)  # ToDb.m:22
        dB_VVtt = todbsmall(E_VVt)  # ToDb.m:23
        dB_VV = np.fft.fftshift(dB_VVtt)  # ToDb.m:24
        e_hh = eehh[:, newteta-1, newphi-1]  # ToDb.m:26
        E_HHt = np.fft.fft(e_hh)  # ToDb.m:27
        dB_HHtt = todbsmall(E_HHt)  # ToDb.m:28
        dB_HH = np.fft.fftshift(dB_HHtt)  # ToDb.m:29
        s1 = a[0: Nphi - 1, 2]  # ToDb.m:31
        s2 = a[0: Nteta * Nphi: Nphi - 1, 1]  # ToDb.m:32
    elif range_freq_theta_phi == 2:
        # RCS_vs_Freq
        e_vv = eevv[:, newteta-1, newphi-1]  # ToDb.m:35
        dB_VV = todbsmall(e_vv)  # ToDb.m:36
        e_hh = eehh[:, newteta-1, newphi-1]  # ToDb.m:38
        dB_HH = todbsmall(e_hh)  # ToDb.m:39
        s1 = a[0: Nphi-1, 2]  # ToDb.m:41
        s2 = a[0: Nteta * Nphi: Nphi, 1]  # ToDb.m:42
    elif range_freq_theta_phi == 3:
        # RCS_vs_Theta
        e_vv = eevv[newfrequency-1, :, newphi-1]  # ToDb.m:45
        dB_VV = todbsmall(e_vv)  # ToDb.m:46
        e_hh = eehh[newfrequency-1, :, newphi-1]  # ToDb.m:48
        dB_HH = todbsmall(e_hh)  # ToDb.m:49
        s1 = a[0: Nfreq * Nteta * Nphi: Nphi * Nteta, 0]  # ToDb.m:51
        s2 = a[0: Nphi-1, 2]  # ToDb.m:52
    elif range_freq_theta_phi == 4:
        # RCS_vs_Phi
        e_vv = eevv[newfrequency-1, newteta-1, :]  # ToDb.m:55
        dB_VV = todbsmall(e_vv)  # ToDb.m:56
        e_hh = eehh[newfrequency-1, newteta-1, :]  # ToDb.m:58
        dB_HH = todbsmall(e_hh)  # ToDb.m:59
        s1 = a[0: Nfreq * Nteta * Nphi: Nphi * Nteta, 0]  # ToDb.m:61
        s2 = a[0: Nteta * Nphi: Nphi, 1]  # ToDb.m:62
    return dB_VV, dB_HH, s1, s2
